### PR TITLE
Completed view-details-back-end feature

### PR DESCRIPTION
**Pull request by MattWotus**
Screen Shot 2020-09-16 at 4 49 57 PM

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -35,6 +35,31 @@ app.get('/api/products', (req, res, next) => {
...
+    where "productId" = $1
+    `;
+  const params = [productId];
+  db.query(sql, params)
+    .then(result => {
+      const product = result.rows[0];
+      if (!product) {
+        next(new ClientError(`Cannot find product with "productId" ${productId}`, 404));
+      } else {
+        return res.status(200).json(result.rows);
```
Comment (scbowler) [inline server/index.js]: should return `product` here not `result.rows` that will put your product in an array which will suck for your front end.

**Review by scbowler — CHANGES_REQUESTED** [review]
One minor change. See comment.

File: server/index.js (hunk 1 above)
Comment (MattWotus) [reply server/index.js]: Fixed!

**Review by scbowler — APPROVED** [review]
Much Better! 👍 👍

**Pull request merged**